enarx-keep-sev-shim/src/asm.rs

```diff
@@ -1,11 +1,23 @@
 // SPDX-License-Identifier: Apache-2.0
 
-//! Declaration of extern assembler instructions
-//!
-//! Once stable rust has native asm!() this will go away
+//! Functions needing `asm!` blocks
 
-extern "C" {
-    pub fn _enarx_asm_triple_fault() -> !;
-    pub fn _rdfsbase() -> u64;
-    pub fn _wrfsbase(val: u64);
+/// Provoke a triple fault to shutdown the machine
+///
+/// An illegal IDT is loaded with limit=0 and an #UD is produced
+///
+/// Fun read: http://www.rcollins.org/Productivity/TripleFault.html
+pub unsafe fn _enarx_asm_triple_fault() -> ! {
+    use x86_64::instructions::tables::lidt;
+    use x86_64::structures::DescriptorTablePointer;
+    // Create an invalid DescriptorTablePointer with no base and limit
+    let dtp = DescriptorTablePointer { limit: 0, base: 0 };
+    // Load the invalid IDT
+    lidt(&dtp);
+    // Provoke an #UD, which will lead to a triple fault, because of the invalid IDT
+    asm!("ud2", options(nomem, nostack));
```

Comment:
Can't we use `unreachable!()` instead here?

Reply:
yeah, makes sense

Reply:
hmm, no we can't ... unreachable!() calls panic before the `ud2`.

Reply:
ack